airflow/providers/weaviate/hooks/weaviate.py

```diff
@@ -201,34 +206,77 @@ def delete_collections(
+                    if if_error == "continue":
+                        self.log.error(e)
+                        failed_collection_list.append(collection_name)
+                    elif if_error == "stop":
+                        raise e
+
+        if if_error == "continue":
+            return failed_collection_list
+        return None
+
+    def delete_by_property(
+        self,
+        collection_name: list[str] | str,
+        by_property: list[str] | str,
+        contains_any: list[Any] | str | None = None,
+        contains_all: list[Any] | str | None = None,
+    ):
+        """
+        Delete collections based on property names.
+
+        :param collection_name: list of collection names or a single collection name to be deleted.
+        :param by_property: property or list of properties to filter documents by.
+        :param contains_any: list of values or a single value.
+            Deletes collections where any of the specified values are present in the property.
+        :param contains_all: list of values or a single value.
+            Deletes collections  where all specified values are present in the property.
+        """
         client = self.get_conn()
-        collection_names = (
-            [collection_names] if collection_names and isinstance(collection_names, str) else collection_names
-        )
+        by_property = [by_property] if isinstance(by_property, str) else by_property
+        contains_any = [contains_any] if isinstance(contains_any, str) else contains_any
+        contains_all = [contains_all] if isinstance(contains_all, str) else contains_all
 
-        failed_collection_list = []
+        collection_names = [collection_name] if isinstance(collection_name, str) else collection_name
         for collection_name in collection_names:
             try:
-                for attempt in Retrying(
-                    stop=stop_after_attempt(3),
-                    retry=(
-                        retry_if_exception(lambda exc: check_http_error_is_retryable(exc))
-                        | retry_if_exception_type(REQUESTS_EXCEPTIONS_TYPES)
-                    ),
-                ):
-                    with attempt:
-                        self.log.info(attempt)
-                        client.collections.delete(collection_name)
+                collection = client.collections.get(collection_name)
+                for prop in by_property:
+                    if contains_any:
+                        collection.data.delete_many(where=Filter.by_property(prop).contains_any(contains_any))
+                    elif contains_all:
+                        collection.data.delete_many(where=Filter.by_property(prop).contains_all(contains_all))
+                    else:
+                        collection.data.delete_many(
+                            where=Filter.by_property(prop).like(collection_name + "*")
+                        )
             except Exception as e:
```

Comment:
What would be the expected Exception of these API calls? would it be possible for us to narrow this exception?

Reply:
@Lee-W Any exceptions caused by `client.collections.get(collection_name)`
Should I commit?
```
 except requests.exceptions.ConnectionError as e:
        raise e
```

Reply:
Yes, this is perfect!